core/src/main/java/io/grpc/MessageEncoding.java

```diff
@@ -39,12 +45,13 @@ public OutputStream compress(OutputStream os) throws IOException {
     }
   };
 
-  private static final ConcurrentMap<String, Decompressor> decompressors =
+  private static final ConcurrentMap<String, DecompressorInfo> decompressors =
       initializeDefaultDecompressors();
 
   /**
    * Represents a message compressor.
    */
+  @ExperimentalApi
```

Comment:
I added it to everything that was public.  This duplicates it in our code, but makes it very easy for clients to find out it isn't ready for general use.